polls/utils.py

```diff
@@ -14,11 +18,17 @@ def batch_selector():
 
 def present_task_for_user(task):
     """mock function"""
+    audios = [task.reference_url, task.transform_url]
     experiment_type = task.batch.experiment_type
-    experiment_type = get_object_or_404(
-        ExperimentTypeTaskPresentation, experiment_type=task.batch.experiment_type
+    task_presentations = ExperimentTypeTaskPresentation.objects.filter(
+        experiment_type=experiment_type
     )
-    return task.reference_url, task.transform_url, experiment_type.task_presentation
+    task_presentation_chosen = random.choice(task_presentations).task_presentation
+    unique_chars = set(task_presentation_chosen)
+    audio_mapping = {}
+    for index, key in enumerate(unique_chars):
+        audio_mapping[key] = audios[index]
+    return audio_mapping, task_presentation_chosen
```

Comment:
This is written in a really tricky way and TBH it's hard to prove this is correct. Especially given that there are ZERO comments and a later programmer might not understand it. I would prefer unit tests. And I would especially prefer that this were written in a really obvious easy-to-understand way that is easy to read and grok.

Reply:
Maybe you should add return types so this is more obvious

Why even return a dict audio_mapping? Why not just shuffle the URLs in the right order? So the functionality is totally contained in this method and encapsulated nicely and harder to break or mess up for another client?

Reply:
Ok let me try to refactor it in some easy way.

Reply:
> Maybe you should add return types so this is more obvious
> 
> Why even return a dict audio_mapping? Why not just shuffle the URLs in the right order? So the functionality is totally contained in this method and encapsulated nicely and harder to break or mess up for another client?

Actually we need to map those to know which audio refers which character like one audio will be bound to character "A" and the other one to "B". Now we can shuffle them according to task_presentation easily which could be anything like "AAB" or "BAB" etc